account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -0,0 +1,158 @@
+import type {
+  EntryPointDef,
+  SmartAccountSigner,
+  SmartContractAccountWithSigner,
+  ToSmartContractAccountParams,
+} from "@aa-sdk/core";
+import {
+  createBundlerClient,
+  getAccountAddress,
+  getEntryPoint,
+  toSmartContractAccount,
+} from "@aa-sdk/core";
+import {
+  concatHex,
+  encodeFunctionData,
+  getContract,
+  type Address,
+  type Chain,
+  type Hex,
+  type Transport,
+} from "viem";
+import { accountFactoryAbi } from "../abis/accountFactoryAbi.js";
+import { addresses } from "../utils.js";
+import { standardExecutor } from "../../msca/account/standardExecutor.js";
+import { singleSignerMessageSigner } from "../modules/single-signer-validation/signer.js";
+
+export const DEFAULT_OWNER_ENTITY_ID = 0;
+
+export type SMAV2Account<
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = SmartContractAccountWithSigner<"SMAV2Account", TSigner, "0.7.0">;
+
+export type CreateSMAV2AccountParams<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner,
+  TEntryPointVersion extends "0.7.0" = "0.7.0"
+> = Pick<
+  ToSmartContractAccountParams<
+    "SMAV2Account",
+    TTransport,
+    Chain,
+    TEntryPointVersion
+  >,
+  "transport" | "chain"
+> & {
+  signer: TSigner;
+  salt?: bigint;
+  factoryAddress?: Address;
+  initCode?: Hex;
+  initialOwner?: Address;
```

Comment:
how is this used? if it's for deferred execution, we could just require that `initCode` be required when using defered execution.

It might be helpful to understand what params need to be passed in for:
1. creating an instance for the root owner
2. creating an instance for a session key signer

Reply:
Addresses are derived from 1 of: account address, or factory + initcode, or factory + owner + salt + entityid. initcode is a transformation of owner + salt + entityId, so assuming a dapp chooses a fixed salt/entityId for users they'll be equivalent. I guess that means we can deprecate the use of ownerAddress?

Reply:
We default to `entityId = 0` if no entityId is specified. It's not good to do this, but it's possible for someone to create an account with the owner associated to some `entityId != 0`. Because of this, I think we don't want to apply union types here